### PR TITLE
LPS-181652 Can't see Knowledge Base Article Child when Parent Article still pending

### PR DESCRIPTION
**Pull request by manuelecastro**
Related issue: [LPS-181652](https://issues.liferay.com/browse/LPS-181652)

**Motivation:** when a user creates a child article where the parent doesn't have an approved status, it's not possible to see its content page. 

**Proposed solution:** This happens because when the `getParentKBArticle()` method is called, it requests the KBA by approved status only, so when the parent article isn't _approved_ it throws a `NoSuchArticleException`. To solve this problem, I proposed to change the status parameter from `WorkflowConstants.STATUS_APPROVED` to `WorkflowConstants.STATUS_ANY` so no that matter what the parent's status is, it will return the article.

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by manuelecastro** [comment]
ci:reevaluate:1377401_10715

**Comment by AliciaGarciaGarcia** [comment]
Hi @manuelecastro 

We think that this not an issue, this should be the expected behavior.
The parent article works as a container; if you cannot see the container (parent), you shouldn't see the content (child articles).

If you have questions with this, you can reach us on the #t-dxp-lima slack channel.

Thanks for all

**Comment by manuelecastro** [comment]
Hi @AliciaGarciaGarcia !

I believe I expressed myself wrong, sorry about that. I meant that it IS possible to see the parent's content but it's **not** possible to see the **child's** content page. 

Here is a video of the bug being reproduced. What do you think?

Thanks! :)

https://user-images.githubusercontent.com/72988896/233180952-fc0177d2-3199-422b-95e8-73a9b76a584d.mp4

**Review by AliciaGarciaGarcia — CHANGES_REQUESTED** [review]

File: modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/model/impl/KBArticleImpl.java [hunk 1]
```diff
@@ -164,7 +164,7 @@ public KBArticle getParentKBArticle() throws PortalException {
 		}
 
 		return KBArticleLocalServiceUtil.getLatestKBArticle(
-			parentResourcePrimKey, WorkflowConstants.STATUS_APPROVED);
+			parentResourcePrimKey, WorkflowConstants.STATUS_ANY);
```
[the remark this answers is not in the document]
Comment (manuelecastro) [reply modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/model/impl/KBArticleImpl.java:167]: @liferay-lima Hi! 

I'm a little bit lost on what kind of approach I could take here. What did Alicia meant when she said the problem is more on the "vertical panel"? Do you know the files that I could look for to make this fix?

Thanks!

[the remark this answers is not in the document]
Comment (adolfopa) [reply modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/model/impl/KBArticleImpl.java:167]: Hey @manuelecastro 
Sorry for the lack of response, but we have some urgent (and unplanned) tasks in our hands right now. Would it be possible to put this on hold for a while until we have more time to look into this?
Thanks!

[the remark this answers is not in the document]
Comment (adolfopa) [reply modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/model/impl/KBArticleImpl.java:167]: @manuelecastro As @AliciaGarciaGarcia said, this fix should be done only for the side panel (there are other usages of that method that expect approved items).

The simplest fix that should work is simply to inline the `getParentKBArticle` code in the `_getParentNode` method. Something like:
```
    private String _getParentNodeURL(KBArticle kbArticle)
		throws PortalException {

		long parentResourcePrimKey = kbArticle.getParentResourcePrimKey();

		if ((parentResourcePrimKey <= 0) ||
			(kbArticle.getParentResourceClassNameId() !=
			 	kbArticle.getClassNameId())) {

			return _createKBFolderRenderURL(kbArticle.getKbFolderId());
		}

		KBArticle parentKBArticle =
			KBArticleLocalServiceUtil.getLatestKBArticle(
				parentResourcePrimKey, WorkflowConstants.STATUS_ANY);

		return _createKBArticleRenderURL(parentKBArticle);
	} 
```

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Comment by diogomelo09** [comment]
ci:close

**Comment by diogomelo09** [comment]
Hi @liferay-lima ! 
Since @manuelecastro is on another team, I'm now handling this ticket for her. 
I resent the PR with the asked modifications here: https://github.com/liferay-lima/liferay-portal/pull/4273

**Pull request closed without merging**